fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/SynchronizedSessionTest.java

```diff
@@ -167,74 +173,69 @@ public void runnersOnSameSessionShouldWorkAndExpireTogether() throws Interrupted
+        try (SynchronizedSessionRunner session1Runner1 = database.newRunner().startSynchronizedSession(lockSubspace1, 2_000)) {
             checkActive(session1Runner1);
 
             UUID session1Id = session1Runner1.getSessionId();
-            try (SynchronizedSessionRunner session1Runner2 = database.newRunner().toSynchronized(lockSubspace, session1Id, 4_000)) {
-                checkActive(session1Runner2);
-                checkActive(session1Runner1);
+            try (SynchronizedSessionRunner session1Runner2 = database.newRunner().joinSynchronizedSession(lockSubspace1, session1Id, 3_000)) {
+                Thread run1 = new Thread(() -> checkActive(session1Runner1));
+                Thread run2 = new Thread(() -> checkActive(session1Runner2));
 
-                AtomicBoolean start = new AtomicBoolean(false);
-                CompletableFuture<Void> session1Runner1Job = session1Runner1.runAsync(context -> AsyncUtil.whileTrue(
-                        () -> CompletableFuture.completedFuture(start.get()),
-                        context.getExecutor()));
-                CompletableFuture<Void> session1Runner2Job = session1Runner1.runAsync(context -> AsyncUtil.whileTrue(
-                        () -> CompletableFuture.completedFuture(start.get()),
-                        context.getExecutor()));
-                database.newRunner().asyncToSync(null, CompletableFuture.allOf(session1Runner1Job, session1Runner2Job));
+                ConcurrentLinkedQueue<Throwable> exceptions = new ConcurrentLinkedQueue<>();
+                Thread.setDefaultUncaughtExceptionHandler((thread, exception) -> exceptions.add(exception));
 
-                // Make the the jobs run almost at the same time with unknown order.
-                start.set(true);
+                run1.start();
+                run2.start();
+                run1.join();
+                run2.join();
 
-                // Runner 1 set the lease end time to 2 seconds in the future, while Runner 2 set it to 4 seconds,
-                // the later one should be honoured. So a new session shouldn't take the lock until 4 seconds.
+                assertTrue(exceptions.isEmpty());
+
+                // Runner 1 set the lease end time to 2 seconds in the future, while Runner 2 set it to 3 seconds,
+                // the later one should be honoured. So a new session shouldn't take the lock until 3 seconds.
                 Thread.sleep(2_000 + 100);
-                assertFailedStartSession(lockSubspace);
+                assertFailedStartSession(lockSubspace1);
 
-                Thread.sleep(2_000);
-                try (SynchronizedSessionRunner session2Runner = newRunnerStartSession(lockSubspace)) {
+                Thread.sleep(1_000);
+                try (SynchronizedSessionRunner session2Runner = newRunnerStartSession(lockSubspace1)) {
                     checkActive(session2Runner);
```

Comment:
Should session 2 be ended to avoid it interacting poorly with other tests (in other words, it might continue writing to `lockSubspace2`, or something)?